src/comment.ts

```diff
@@ -113,20 +138,49 @@ export const handleComment: ActionMapInput = async (
     ...mRule,
     blobURL: getBlobURL(mRule.path, files, owner, repo, base),
   }));
+
   const commentsFromRules = composeCommentsForUsers(rulesWithBlobURL);
+
   const rawComments = await getAllComments(client, {
     owner,
     repo,
     issue_number: prNumber,
   });
-  const comments = rawComments.map(({ body }) => body);
-
-  const onlyNewComments = commentsFromRules.filter((comment: string) => !comments.includes(comment));
 
-  debug('comments to add:', onlyNewComments);
+  // Filter existing comments by USE_HERALD_ACTION tag (HTML comment) and key by path
+  const useHeraldActionComments = rawComments
+    .filter(({ body }) => USE_HERALD_ACTION_TAG_REGEX.test(body.split('\n')[0]))
+    .reduce((memo, comment) => {
+      const bodyFirstLine = comment.body.split('\n')[0];
+      const path = bodyFirstLine.replace('<!-- USE_HERALD_ACTION ', '').replace(' -->', '');
```

Comment:
I'm wondering if instead, you can have a regex that only matches what is between those replace you are doing.